### ISSUE TITLE
Issue with Pupeteer launching Chrome

### ISSUE THREAD

**Issue by rocketLeagueTraderJoe** [jamieweavis/dodgem#7]
I tried using this on Ubuntu 18.04 LTS on a virtual machine. Node.js version is v8.10.0
After I start dodgem and enter the interval I get this message:
```
(node:1546) UnhandledPromiseRejectionWarning: Error: Failed to launch chrome!
[0515/072301.058458:ERROR:icu_util.cc(133)] Invalid file descriptor to ICU data received.
[0515/072301.062008:FATAL:content_main_delegate.cc(57)] Check failed: false. 
#0 0x56526799482c base::debug::StackTrace::StackTrace()
#1 0x56526791b290 logging::LogMessage::~LogMessage()
#2 0x565265bcbde3 content::ContentMainDelegate::TerminateForFatalInitializationError()
#3 0x56526764a941 content::ContentMainRunnerImpl::Initialize()
#4 0x565267654c12 service_manager::Main()
#5 0x565267649184 content::ContentMain()
#6 0x56526b821a39 headless::(anonymous namespace)::RunContentMain()
#7 0x56526b821ac2 headless::HeadlessBrowserMain()
#8 0x565267651f8f headless::HeadlessShellMain()
#9 0x565265bca1ac ChromeMain
#10 0x7fd149309b97 __libc_start_main
#11 0x565265bca02a _start


TROUBLESHOOTING: https://github.com/GoogleChrome/puppeteer/blob/master/docs/troubleshooting.md

    at onClose (/usr/local/lib/node_modules/dodgem/node_modules/puppeteer/lib/Launcher.js:255:14)
    at Interface.helper.addEventListener (/usr/local/lib/node_modules/dodgem/node_modules/puppeteer/lib/Launcher.js:244:50)
    at emitNone (events.js:111:20)
    at Interface.emit (events.js:208:7)
    at Interface.close (readline.js:370:8)
    at Socket.onend (readline.js:149:10)
    at emitNone (events.js:111:20)
    at Socket.emit (events.js:208:7)
    at endReadableNT (_stream_readable.js:1064:12)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
(node:1546) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:1546) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
It looks like Pupeteer is having some issues starting up chrome. I checked for the Pupeteer dependencies and they're all there. Other than that, I have no idea what the issue might be. If it changes things at all, I installed dodgem with `sudo yarn global add dodgem`, with yarn being v1.6.0

Also if I do this:
`/usr/local/share/.config/yarn/global/node_modules/puppeteer/.local-chromium/linux-555668/chrome-linux$ ./chrome`

I still get:
```
[0515/081740.699389:ERROR:icu_util.cc(133)] Invalid file descriptor to ICU data received.
[0515/081740.699473:FATAL:content_main_delegate.cc(57)] Check failed: false. 
#0 0x5563f0e6882c base::debug::StackTrace::StackTrace()
#1 0x5563f0def290 logging::LogMessage::~LogMessage()
#2 0x5563ef09fde3 content::ContentMainDelegate::TerminateForFatalInitializationError()
#3 0x5563f0b1e941 content::ContentMainRunnerImpl::Initialize()
#4 0x5563f0b28c12 service_manager::Main()
#5 0x5563f0b1d184 content::ContentMain()
#6 0x5563ef09e1b3 ChromeMain
#7 0x7f0272922b97 __libc_start_main
#8 0x5563ef09e02a _start

Aborted (core dumped)
```

But when I do this:
`/usr/local/share/.config/yarn/global/node_modules/puppeteer/.local-chromium/linux-555668/chrome-linux$ sudo ./chrome --no-sandbox --disabled-setuid-sandbox`

I get a usable chrome window pop up and this is in the terminal:
```
[2082:2170:0515/082140.333525:ERROR:bus.cc(394)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[2082:2082:0515/082140.461890:ERROR:gpu_process_transport_factory.cc(1012)] Lost UI shared context.

(chrome:2082): LIBDBUSMENU-GLIB-WARNING **: 08:21:40.562: Unable to get session bus: Unknown or unsupported transport “disabled” for address “disabled:”
[2175:2191:0515/082140.652312:ERROR:command_buffer_proxy_impl.cc(114)] ContextResult::kFatalFailure: Shared memory handle is not valid
[0515/082148.982805:ERROR:nacl_helper_linux.cc(310)] NaCl helper process running without a sandbox!
Most likely you need to configure your SUID sandbox correctly
```

**Comment by jamieweavis** [comment]
It seems like the problem could be permission related - potentially because you installed `dodgem` with `sudo`. Please try fixing the permissions of Puppeteer by following the steps stated in GoogleChrome/puppeteer#2519.

**Comment by rocketLeagueTraderJoe** [comment]
I also tried using npm but was getting an error: EACCES. So I did the whole thing over again, except I changed the Default Directory, as per [here](https://docs.npmjs.com/getting-started/fixing-npm-permissions). I installed Dodgem again and this time it was ok.

Strangely enough while I was trying to figure the above out, I tried the exact same thing with Debian 9.4. For Debian, it still wouldn't work until I edited Dodgem.js from:

`const browser = await puppeteer.launch({
      ignoreHTTPSErrors: true,
      headless: this.headless
    });`

to

`const browser = await puppeteer.launch({
      ignoreHTTPSErrors: true,
      headless: this.headless,
      args: ['--no-sandbox', '--disable-setuid-sandbox']
    });`

But anyways, seems to work fine on Ubuntu (not really sure what was happening on Debian). Thanks!
